### PR TITLE
Workaround for database services which doesn't support MyISAM

### PR DESCRIPTION
**Pull request by mgomse**
## Changes in this pull request  
This PR adds a workaround for MySQL services which don't support MyISAM engine. It was created mainly for Azure Database for [MySQL service]((https://azure.microsoft.com/en-us/services/mysql/) - https://docs.microsoft.com/en-us/azure/mysql/concepts-limits#unsupported

## Additional info  

This PR should make no changes to the actual installation process - it is just an addition:

* `is-myisam-supported` -cert-path won't be asked interactively
* if `is-myisam-supported` parameter will be omitted in CLI (or env variables) then the installation is like it was until now

Theoretically, it could be done better but then `installerNeedsOption` usage should be rewritten (e.g. currently parameters like `skip-database-data-dump` doesn't work because of this method), so I decided that in one PR should be done only one change.

### CODE REVIEW

File: bundles/InstallBundle/Command/InstallCommand.php [hunk 1]
```diff
@@ -115,6 +115,12 @@ private function getOptions()
                 'default' => 3306,
                 'group' => 'db_credentials',
             ],
+            'is-myisam-supported' => [
+                'description' => 'If set to "no" then MyISAM engine is changed to InnoDB in all tables (useful e.g. in Azure database for MySQL)',
+                'mode' => InputOption::VALUE_REQUIRED,
```
Comment (brusch) [inline bundles/InstallBundle/Command/InstallCommand.php]: Value should be optional, right?

File: bundles/InstallBundle/Command/InstallCommand.php [hunk 2]
```diff
@@ -115,6 +115,12 @@ private function getOptions()
...
+                'default' => 'yes',
```
Comment (brusch) [inline bundles/InstallBundle/Command/InstallCommand.php]: I think we should use a bool here, and call the option `force-innodb-tables`.
Then if the option is set we set the flag accordingly. I think we should then do the same with MEMORY tables. 
Thanks!

**Comment by brusch** [comment]
@mgomse any update?

File: bundles/InstallBundle/Command/InstallCommand.php (hunk 1 above)
Comment (mgomse) [reply bundles/InstallBundle/Command/InstallCommand.php]: Right, fixex

File: bundles/InstallBundle/Command/InstallCommand.php (hunk 2 above)
Comment (mgomse) [reply bundles/InstallBundle/Command/InstallCommand.php]: To be true I'm not sure about it - MyISAM problem is that it has "lack of transaction support which can lead to cases of data loss." - that's why e.g Azure doesn't support it (https://feedback.azure.com/forums/597982-azure-database-for-mysql/suggestions/19271050-add-myisam-engine-support)

However each engine has got its own advantages (probably that's why not all tables are by default InnoDb). So in my opinion we shouldn't have one potion to force all tables to be the same but according to use cases we should have possibility to "turn off" this special functionalities. And in the Azure database use-case we would like to "turn off" only the MyISAM, becasue why we should also abandon MEMORY engine advantages?

@brusch what do you think?

**Comment by mgomse** [comment]
@brusch I'm very sorry that I took so long ... but finally I fixed what I could and I answered where I do not fully agree. Please let me know what do you think about it?

File: bundles/InstallBundle/Command/InstallCommand.php (hunk 2 above)
Comment (brusch) [reply bundles/InstallBundle/Command/InstallCommand.php]: @mgomse The MEMORY storage engine also doesn't support transactions 😉
See: https://mariadb.com/kb/en/library/show-engines/

Comment (brusch) [reply bundles/InstallBundle/Command/InstallCommand.php]: But we're going away from the MEMORY engine anyway, see: #4901

Comment (brusch) [reply bundles/InstallBundle/Command/InstallCommand.php]: But please, let's rename the flag and make it a boolean, then we can merge it. Thanks!

**Comment by brusch** [comment]
Obsolete, see #5218

**Pull request closed without merging**